pal/inc/pal_char16.h

```diff
@@ -28,26 +28,22 @@ This file is used to define the wchar_t type as a 16-bit type on Unix.
 // vc++, for whom wchar_t is already a typedef instead of a built-in.
 
 #if !defined(PAL_STDCPP_COMPAT) || defined(USING_PAL_STDLIB)
-#if defined (PLATFORM_UNIX) && defined(__GNUC__)
-#undef wchar_t
-#define wchar_t __wchar_16_cpp__
```

Comment:
What's the reason to make changes in this file? Original code was to redefine wchar_t. What doesn't work?